lib/index.js

```diff
@@ -81,30 +80,26 @@ const readOutOfTreeIgnoreFiles = (root, rel, result = []) => {
 }
 
 class PackWalker extends IgnoreWalker {
-  constructor (opts) {
+  constructor (tree, opts) {
     const options = {
       ...opts,
       includeEmpty: false,
       follow: false,
-    }
-
-    // we path.resolve() here because ignore-walk doesn't do it and we want full paths
-    options.path = resolve(options.path || process.cwd()).replace(/\\/g, '/')
-
-    if (!options.ignoreFiles) {
-      options.ignoreFiles = [
+      // we path.resolve() here because ignore-walk doesn't do it and we want full paths
+      path: resolve(opts?.path || tree.path).replace(/\\/g, '/'),
+      ignoreFiles: opts?.ignoreFiles || [
         defaultRules,
         'package.json',
         '.npmignore',
         '.gitignore',
         strictRules,
-      ]
+      ],
     }
 
     super(options)
     this.isPackage = options.isPackage
     this.seen = options.seen || new Set()
-    this.tree = options.tree // no default, we'll load the tree later if we need to
+    this.tree = tree
```

Comment:
i think this should have some kind of runtime check, so that if something not-tree-like is passed, it eagerly throws an exception.